### PR TITLE
feat: [PL-25020] Show Inherited entities by user group

### PR DESCRIPTION
**Pull request by reetika04**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by upadhyay-an** [comment]
create_image PL-25020

**Comment by reetika04** [comment]
dockerize PL-23734-v4

**Comment by upadhyay-an** [comment]
dockerize PL-23734

**Comment by upadhyay-an** [comment]
dockerize PL-23734-v6

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const { orgName, projectName } = row.original
+  const { getString } = useStrings()
+
+  if (!projectName) {
+    return <Text>{orgName}</Text>
+  }
+
+  return (
+    <>
+      <Text>{projectName}</Text>
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:48]: color in text

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  )
+}
+
+const ScopeList: React.FC = () => {
+  const { getString } = useStrings()
+  const { accountId, orgIdentifier, userGroupIdentifier } =
+    useParams<PipelineType<ProjectPathProps & { userGroupIdentifier: string }>>()
+  const { data, loading, error } = useGetInheritingChildScopeList({
+    identifier: userGroupIdentifier,
+    queryParams: {
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:62]: use getUserGroupQueryParams

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+      width: '75%',
+      accessor: 'projectName',
+      Cell: RenderScopeDescription
+    }
+  ]
+
+  if (loading)
+    return (
+      <Icon
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:85]: <PageSpinner />

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        name="steps-spinner"
+        size={32}
+        color={Color.GREY_600}
+        style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}
+        padding="large"
+      />
+    )
+
+  if (error) return <Text color={Color.RED_400}>{getString('rbac.inheritedScope.errorMessage')}</Text>
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:94]: Render the error message from the api instead.

Comment (reetika04) [reply src/modules/20-rbac/components/ScopeList/ScopeList.tsx:94]: Please use `<PageError />`

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+  return scopeListResponse && scopeListResponse.length ? (
+    <TableV2<ScopeName> columns={columns} data={scopeListResponse} hideHeaders={true} />
+  ) : (
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:98]: No data state should match the mocks.

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx [hunk 6]
```diff
@@ -250,12 +254,20 @@ const UserGroupDetails: React.FC = () => {
...
-          <Layout.Vertical spacing="medium">
+          <Layout.Vertical spacing="medium" padding={{ bottom: 'large' }}>
             <Text color={Color.BLACK} font={{ size: 'medium', weight: 'bold' }}>
               {getString('common.notificationPreferences')}
             </Text>
             <NotificationList userGroup={userGroup} inherited={userGroupInherited} onSubmit={refetch} />
           </Layout.Vertical>
+          {INHERITED_USER_GROUP && (
+            <Layout.Vertical spacing="large" border={{ top: true }}>
+              <Text color={Color.BLACK} font={{ size: 'medium', weight: 'bold' }} padding={{ top: 'large' }}>
```
Comment (reetika04) [inline src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:265]: use font variation.

File: src/modules/20-rbac/pages/UserGroupDetails/__tests__/mock.ts [hunk 7]
```diff
@@ -170,3 +175,27 @@ export const userGroupCheckingInfo = {
...
+
+export const inheritingChildScopeListData: { data: ResponseListScopeName } = {
+  data: {
+    status: 'SUCCESS',
+    data: [
+      {
+        accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
+        orgName: 'checking org',
+        orgIdentifier: 'checking_org',
+        projectName: undefined,
```
Comment (reetika04) [inline src/modules/20-rbac/pages/UserGroupDetails/__tests__/mock.ts:187]: You don't need this in mock data

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx [hunk 8]
```diff
@@ -250,12 +254,20 @@ const UserGroupDetails: React.FC = () => {
...
+                {getString('rbac.inheritedScope.title')}
```
Comment (reetika04) [inline src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:266]: Please use different titles for account and org
Org scoped UserGroup should only say Projects using this group

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    <>
+      <Text>{projectName}</Text>
+      <Text font={{ size: 'small' }} padding={{ top: 'xsmall' }} color={Color.GREY_400}>
+        {getString('rbac.inheritedScope.projectOrg', { orgName })}
+      </Text>
+    </>
+  )
+}
+
+const ScopeList: React.FC = () => {
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:56]: UserGroupReferenceScopeList

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        padding="large"
+      />
+    )
+
+  if (error) return <Text color={Color.RED_400}>{getString('rbac.inheritedScope.errorMessage')}</Text>
+
+  return scopeListResponse && scopeListResponse.length ? (
+    <TableV2<ScopeName> columns={columns} data={scopeListResponse} hideHeaders={true} />
+  ) : (
+    <Text>{getString('rbac.inheritedScope.noData')}</Text>
```
Comment (reetika04) [inline src/modules/20-rbac/components/ScopeList/ScopeList.tsx:99]: Same.

File: src/modules/20-rbac/components/ScopeList/ScopeList.tsx (hunk 5 above)
Comment (naman-goenka) [reply src/modules/20-rbac/components/ScopeList/ScopeList.tsx:98]: No data text should change according to scope 
`No project inheriting this UserGroup`

Comment (naman-goenka) [reply src/modules/20-rbac/components/ScopeList/ScopeList.tsx:98]: We can also add pagination similar to users section in usergroup details.

File: src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx (hunk 6 above)
Comment (upadhyay-an) [reply src/modules/20-rbac/pages/UserGroupDetails/UserGroupDetails.tsx:265]: Done but wanted to highlight following two points

1. Figma mocks are not consistent with font-variation and hence title heading would look a bit smaller.
2. In order to keep code and design consistent I've changed it for all panel headings.

**Comment by avinashmadhwani02** [comment]
dockerize

**Comment by naman-goenka** [comment]
dockerize [PL-23734](https://harness.atlassian.net/browse/PL-23734)-v7

**Pull request closed without merging**